### ISSUE TITLE
Relationship belongsToMany doesnt work.

### ISSUE THREAD

**Issue by Rukhsar** [sunel/eav#10]
Hi,

I have used your package to complete my project and done almost whole of my project. Just found that when I create a Product Model and it extends your Eav\Model. If I put a belongstoMany relationship with another model which is extending Laravel model it doesn't return anything. I know you have already extended the Eloquent Model and all relationship should be working fine but it's not.

Any idea? If you can guide me on this.

Thanks

**Comment by sunel** [comment]
Sure, can you post the code

**Comment by Rukhsar** [comment]
Hi,

Bike Model: 

```
use Eav\Model;

class Bike extends Model
{
    const ENTITY  = 'bike';

    public function categories()
    {
        return $this->belongsToMany(Category::class, 'bike_category', 'bike_id', 'category_id')->withTimestamps();
    }
}
```
Category Model:

```
class Category extends Model
{
    public function bikes()
    {
        return $this->belongsToMany(Bike::class, 'bike_category', 'bcategory_id', 'bike_id')->withTimestamps();
    }
}

```

No if I do like this.

```
$category = Category::findOrFail(2);
$bikes = $category->bikes;

```

It breaks and If I change `use Eav\Model;` in Bike Model to `use Illuminate\Database\Eloquent\Model;` then I do get the bikes for category id 2 but when using Eav\Model this relationship not working.

Thanks

**Comment by sunel** [comment]
@Rukhsar 

I believe you have set the `primaryKey `

```php
use Eav\Model;

class Bike extends Model
{
    const ENTITY  = 'bike';

    protected $primaryKey = 'bike_id';

    public function categories()
    {
        return $this->belongsToMany(Category::class, 'bike_category', 'bike_id', 'category_id')->withTimestamps();
    }
}

class Category extends Model
{
  
    protected $primaryKey = 'category_id';

    public function bikes()
    {
        return $this->belongsToMany(Bike::class, 'bike_category', 'category_id', 'bike_id')->withTimestamps();
    }
}

```

And also the latest code of the eav , as i have made changes for using custom `primaryKey `.

**Comment by Rukhsar** [comment]
I dont think so its primary key as I am using `id` as primary key. Get following error

`SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'from `bikes` inner join `bike_category` on `bikes`.`id` = `bike_category`.`bike_' at line 1 (SQL: select from `bikes` inner join `bike_category` on `bikes`.`id` = `bike_category`.`bike_id` where `bike_category`.`bcategory_id` = 2)`

**Comment by sunel** [comment]
@Rukhsar 

This was a bug, i have fixed and pushed the code 

Tested the below scenarios

```php
$category = App\Category::findOrFail(1);
$products = $category->products;

$category = App\Category::findOrFail(1);
$products = $category->products()->select(['attr.*'])->get();

$product = App\Products::findOrFail(23);
$categories = $product->categories;

$product = App\Products::select(['attr.*'])->find(23);
$categories = $product->categories;
``` 

Works as expected.

Update to lasted commit and test it out

> Sorry for not tagging any release as i need to do more testing.

**Comment by sunel** [comment]
@Rukhsar 

Found few more issue after the code change, do not pull the code until further notice.

**Comment by sunel** [comment]
@Rukhsar 

Fixed few more issue, you can pull the code and test if your relation works as expected.
